### PR TITLE
[FIXED JENKINS-32831] do not fail if there are no metadata for tool installers

### PR DESCRIPTION
**Pull request by varmenise**
Before JENKINS-32328 the tool installers updates were always checked against the default update center.
Now, tools installers are checked against the update centers actually configured. If the experimental update center is the only one configured, there are no json files available for tool installers. So we return the validation error, although the experimental update center has been successfully consulted for what concerns the plugins.
The error message is misleading, the problem is not the signature but the update json files for tool installers not being found.

### CODE REVIEW

**Comment by varmenise** [comment]
@reviewbybees esp @daniel-beck @ndeloof

**Comment by stephenc** [comment]
:bee:

**Comment by ndeloof** [comment]
LGTM

**Comment by mslusarczyk** [comment]
Looks good :bee:

File: core/src/main/java/hudson/model/DownloadService.java [hunk 1]
```diff
@@ -407,13 +409,17 @@ public FormValidation updateNow() throws IOException {
...
                 }
                 jsonList.add(o);
             }
-            if (jsonList.size() == 0) {
-                return FormValidation.warning("None of the Update Sites passed the signature check");
+            if (jsonList.size() == 0 && toolInstallerMetadataExists) {
+                return FormValidation.warning("None of the tool installer metadata passed the signature check");
+            } else if (!toolInstallerMetadataExists) {
+                LOGGER.log(Level.WARNING, "No tool installer metadata found for " + id);
+                return FormValidation.ok();
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/DownloadService.java]: Maybe it should be a warning

Comment (varmenise) [inline core/src/main/java/hudson/model/DownloadService.java]: mmm not really, the experimental update centers for example does not have any and same goes for current and others..

**Comment by varmenise** [comment]
mmm, looks like I have cause a regression, I ll look into it

**Comment by varmenise** [comment]
Reopening to check regression really exists

**Comment by varmenise** [comment]
The current failure is due to the server we use to resolve the IP being down http://freegeoip.net/

**Comment by varmenise** [comment]
@oleg-nenashev I think this can be merged right?

**Comment by oleg-nenashev** [comment]
@varmenise Ready to go IMHO. You know my PoV of squashing, but it's up to you

**Comment by varmenise** [comment]
@oleg-nenashev I haven t squashed cause the typo is from a previous merge, not this PR. But I will do that :)

**Comment by daniel-beck** [comment]
@varmenise Sorry, missed your last comment before merging.

**Comment by varmenise** [comment]
@daniel-beck np cause we took it offline with @oleg-nenashev and realised that since the typo belongs to another merge (not this PR) it's better to leave it in a separate commit.

**Pull request merged**